core/src/main/java/dev/masonroot/nora/config/ConfigManager.java

```diff
@@ -0,0 +1,530 @@
+   * </ul>
+   *
+   * @param config the configuration model instance; must not be null
+   * @param configPath the path to the configuration file; must not be null
+   */
+  public ConfigManager(final T config, final Path configPath) {
+    this(config, configPath, false);
+  }
+
+  /**
+   * Constructs a new {@code ConfigManager} with the specified configuration model, configuration
+   * file path, and file watching option.
+   *
+   * <p><b>Why:</b>
+   *
+   * <ul>
+   *   <li>This constructor initializes the {@code ConfigManager} with a specific configuration
+   *       model, path to the configuration file, and an option to enable file watching for
+   *       automatic reloading of the configuration.
+   *   <li>It ensures that the configuration file is created if it does not exist and loads the
+   *       configuration from the file.
+   * </ul>
+   *
+   * <p><b>Notes:</b>
+   *
+   * <ul>
+   *   <li>The {@code @NonNull} annotation indicates that the parameters should not be null.
+   *   <li>The {@code reload} parameter determines whether file watching is enabled for automatic
+   *       reloading of the configuration.
+   * </ul>
+   *
+   * @param config the configuration model instance; must not be null
+   * @param configPath the path to the configuration file; must not be null
+   * @param reload whether to enable file watching for automatic reloading of the configuration
+   */
+  public ConfigManager(
+      @NonNull final T config, @NonNull final Path configPath, final boolean reload) {
+    this.configPath = configPath;
+    this.config = config;
+    this.create();
+    this.reload();
+    if (reload) {
+      this.startFileWatcher();
+    }
+  }
+
+  /**
+   * Creates the configuration file if it does not exist.
+   *
+   * <p><b>Why:</b>
+   *
+   * <ul>
+   *   <li>This method ensures that the configuration file is created if it does not exist.
+   *   <li>It checks the file path and creates the file if necessary.
+   * </ul>
+   *
+   * <p><b>Notes:</b>
+   *
+   * <ul>
+   *   <li>If the configuration file already exists, it validates that the file is readable and
+   *       writable.
+   *   <li>The method throws an exception if the file path is invalid or the file cannot be created.
+   * </ul>
+   *
+   * <p><b>Performance:</b>
+   *
+   * <ul>
+   *   <li>It avoids redundant file creation by checking the file's existence first.
+   *   <li>The use of `Files.createFile` ensures atomic file creation, reducing the risk of race
+   *       conditions.
+   *   <li>Validating the file's readable and writable upfront prevents potential runtime errors,
+   *       improving overall application stability.
+   * </ul>
+   *
+   * @throws ConfigManagerException if the configuration file cannot be created
+   */
+  public void create() throws ConfigManagerException {
+    if (!this.exists()) {
+      try {
+        Files.createFile(this.configPath);
```

Comment:
I'd argue it reduces the amount of brain power you need when reading the code.

Sure, it may be unnecessary when there isn't a local scope variable with the same name, but it helps me understand that it's referencing a global variable. My eyes naturally look for the local variable declaration when I don't see the this keyword.

It's also super helpful, especially if IDEA's variable renaming feature decides to break and you are forced to use the find and replace feature

Funnily enough, this has happened more than a few times. I have to relaunch IDEA. It's a whole process.